exercicios/para-casa/LohanaR/atividadepracasa.py

```diff
@@ -0,0 +1,132 @@
+        except ValueError:
+            print("Entrada inválida. Por favor, digite um número inteiro.\n")
+
+    return codigo_opcao
+
+def incluir_nova_discente(dataset):
+    nome = input("Por favor, digite o nome da discente: ")
+    try:
+        faltas = int(input("Por favor, digite o número de faltas: "))
+    except ValueError:
+        print("Esse número de faltas é inválido. Por favor, digite um número inteiro.")
+        return
+
+    notas = []
+    print("Por favor, digite as notas da discente. Se desejar encerrar, digite 'finalizar' para concluir.")
+    while True:
+        nota = input("Nota: ")
+        if nota.lower() == 'finalizar':
+            break
+        try:
+            nota_float = float(nota)
+            if 0 <= nota_float <= 10: 
+                notas.append(nota_float)
+            else:
+                print("Essa nota não atende o critério do intervalo aceitável (0 a 10).")
+        except ValueError:
+            print("Essa entrada é inválida. Por favor, digite um número válido ou 'finalizar' para concluir.")
+
+    if len(notas) == 0:
+        print("Discente não incluída.")
+        return
+
+    dataset['discentes'].append({
+        'nome': nome,
+        'faltas': faltas,
+        'notas': notas
+    })
+    print("Discente incluída!")
+
+def consultar_lista_discentes(dataset):
+    if len(dataset['discentes']) == 0:
+        print("\nSem cadastro de discentes.")
+        return
+
+    print("\nDiscentes:")
+    for discente in dataset['discentes']:
+        print(f"Nome: {discente['nome']}, Faltas: {discente['faltas']}, Notas: {discente['notas']}")
+
+def consultar_faltas_discente(dataset):
+    if len(dataset['discentes']) == 0:
+        print("\nSem cadastro de discentes.")
+        return
+
+    nome = input("Por favor, digite o nome da discente: ")
+    for discente in dataset['discentes']:
+        if discente['nome'] == nome:
+            print(f"Discente: {discente['nome']}, Faltas: {discente['faltas']}")
+            return
+    print("Ops, essa discente não consta no sistema.")
+
+def consultar_notas_discente(dataset):
+    if len(dataset['discentes']) == 0:
+        print("\nSem cadastro de discentes.")
+        return
+
+    nome = input("Por favor, digite o nome da discente: ")
+    for discente in dataset['discentes']:
+        if discente['nome'] == nome:
+            print(f"Discente: {discente['nome']}, Notas: {discente['notas']}")
+            return
+    print("Ops, essa discente não consta no sistema.")
+
+def consultar_status_aprovacao(dataset):
+    if len(dataset['discentes']) == 0:
+        print("\nOps, não consta discentes registradas no sistema.")
+        return
+
+    nome = input("Por favor, digite o nome da discente: ")
+    for discente in dataset['discentes']:
+        if discente['nome'] == nome:
+            media = sum(discente['notas']) / len(discente['notas'])
+            presenca = 1 - (discente['faltas'] / 100)
```

Comment:
Esse cálculo de percentual não está correto, pois não recebemos a lista de presença no início. Assim, não temos o número total de aulas, e por fim, não temos a proporção de aulas que a pessoa faltou. O cálculo correto seria:
Percentual_de_faltas = Quantidade_de_faltas / Quantidade_de_aulas